### ISSUE TITLE
Trading System & Limiteds + Groups?

### ISSUE THREAD

**Issue by larben45** [tp-link-extender/MercuryCore#344]
### Describe the problem

It would make revivals more active and enjoyable tbh, Some people have more interest in trading and collectibles than just playing.

### Describe the proposed solution

not sure if these small features will be added but im mentioning them anyway.

1. Change available assets into limited/offsale.

2. limit purchases for stocked limiteds so they don't get hoarded.

4. Discord bot announce whenever a new limited is released, or just a regular message if its a hat, package etc.

(Off topic but it would be cool if it showed unapproved assets in a private discord channel and you can approve them there, no idea if this feature exists)

**Comment by Heliodex** [comment]
Trading and limiteds/collectibles was something I considered heavily while implementing the new [Economy service](https://github.com/tp-link-extender/MercuryCore/tree/main/Economy), with several ideas:

- allowing some types of assets to be created as limited
- having all paid assets be limited, with free assets having an unlimited supply
- requiring a fee proportional to both the price and the supply when creating limited assets, so higher priced or higher supply limiteds would cost more to create
- even removing unlimited assets entirely and allowing anyone to create and trade limited assets

At one point I gave up and implemented a prototype of limiteds as NFTs on blockchain instead. I'm still not sure which solution is the best and how limiteds could be implemented into the database/economy services.

Relating to limited purchases for stocked limiteds, this seems simple at first with a cap on how many of an asset can be purchased, but seems to me to have a few edge cases:

- Would this only happen for before the stock runs out? Would people be able to buy as many resold assets as they wanted?
- If not, what if someone tried to trade a new asset to the owner after they already owned the maximum number?
- Could a user still buy up many assets on multiple alternate accounts and trade them to their main account? Do we need a system to detect or restrict alternate accounts?

Ultimately, I'm not an LMADer or trader so I don't exactly know what those people want. It would be helpful to receive more feedback on how limiteds or trading should work in Mercury Core. I'm all in favour of these features if they're wanted by the community and help to increase the 'fun' factor.

On Discord-related features, these also seem interesting but would probably be added to [tp-link-extender/Bot](https://github.com/tp-link-extender/Bot) instead  of this repository.

**Comment by ui0ppk** [comment]
I do not think NFTs for limiteds are a good idea

**Comment by Heliodex** [comment]
> I do not think NFTs for limiteds are a good idea

Yeah, mainly problems with self hosting and requiring connection to an external network, even if it doesn't have the same issues with latency or reliability, I still found problematic and somewhat dilutes the benefits of easier trading/unified currency. Some of the implementations I prototyped in the Economy service had similar enough behaviour without the same downsides, which pushed me towards going with that instead.
